streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -874,7 +826,53 @@ class ZStream[-R, +E, +A] private[stream] (private[stream] val structure: ZStrea
    * @note Prefer capacities that are powers of 2 for better performance.
    */
   final def bufferSliding(capacity: Int): ZStream[R, E, A] =
-    ZStream.managed(ZManaged.make(Queue.sliding[A](capacity))(_.shutdown)).flatMap(bufferSignal)
+    ZStream {
+      for {
+        as    <- self.process
+        queue <- Queue.sliding[(Take[E, A], Promise[Nothing, Unit])](capacity).toManaged(_.shutdown)
+        start <- Promise.make[Nothing, Unit].toManaged_
+        _     <- start.succeed(()).toManaged_
```

Comment:
I looked through code and it seems we don't do it often enough, but I wonder if there should be `Promise.succeed[Now]` and `Promise.fail[Now]` constructors.